### PR TITLE
Add TargetGroupPolicy docs

### PR DESCRIPTION
**Pull request by solmonk**
**What type of PR is this?**
documentation

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs from aws-gateway-controller showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this PR introduce any new dependencies?**:

**Will this break upgrades or downgrades. Has updating a running cluster been tested?**:

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/configure/https.md [hunk 1]
```diff
@@ -97,3 +97,27 @@ spec:
...
+kind: TargetGroupPolicy
+metadata:
+    name: test-policy
+spec:
+    targetRef:
+        group: ""
+        kind: Service
+        name: my-parking-service # Put service name here
+    protocol: HTTP
+    protocolVersion: HTTPS
```
Comment (mikhail-aws) [inline docs/configure/https.md:117]: https is not a version

File: docs/reference/target-group-policy.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+Attaching TargetGroupPolicy to a resource that is already referenced by a route may result in a replacement
+of VPC Lattice TargetGroup resource.
+
+|Field	|Description	|
+|---	|---	|
+|`apiVersion` *string*	|``application-networking.k8s.aws/v1alpha1``	|
+|`kind` *string*	|``TargetGroupPolicy``	|
+|`metadata` [*ObjectMeta*](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#objectmeta-v1-meta)	|Kubernetes metadata for the resource.	|
+|`spec` *TargetGroupPolicySpec*	|Defines the desired state of TargetGroupPolicy.	|
```
Comment (mikhail-aws) [inline docs/reference/target-group-policy.md:29]: it does not say which fields here can cause replacement

File: docs/reference/target-group-policy.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+## TargetGroupPolicySpec
+
+Appears on: TargetGroupPolicy
+
+TargetGroupPolicySpec defines the desired state of TargetGroupPolicy.
+
+Updates to certain fields of this configuration result in a replacement of VPC Lattice TargetGroup resource.
```
Comment (mikhail-aws) [inline docs/reference/target-group-policy.md:37]: I would avoid vague mentions like "certain fields. It's better to list fields that can cause replacement here, even it's duplicated in the table below, saying something important twice wont hurt.

**Review by mikhail-aws — CHANGES_REQUESTED** [review]

File: docs/configure/https.md (hunk 1 above)
Comment (solmonk) [reply docs/configure/https.md]: updated, thanks

File: docs/reference/target-group-policy.md (hunk 2 above)
Comment (solmonk) [reply docs/reference/target-group-policy.md]: makes a lot of sense, updated this a bit, tried to be more specific on this

**Review by mikhail-aws — APPROVED** [review]

**Pull request merged**